pkg/scaleview/scale_view_helper.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Singularity Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scaleview
+
+import (
+	"github.com/samber/lo"
+
+	risingwavev1alpha1 "github.com/risingwavelabs/risingwave-operator/apis/risingwave/v1alpha1"
+	"github.com/risingwavelabs/risingwave-operator/pkg/consts"
+)
+
+type RisingWaveScaleViewHelper struct {
+	risingwave *risingwavev1alpha1.RisingWave
+	component  string
+}
+
+func findGroup[T any](groups []T, target string, name func(*T) string) (*T, int) {
+	for i, group := range groups {
+		if name(&group) == target {
+			return &groups[i], i
+		}
+	}
+	return nil, 0
+}
+
+func (r *RisingWaveScaleViewHelper) getReplicasPtr(group string) (*int32, int) {
+	switch r.component {
+	case consts.ComponentMeta:
+		if group == "" {
```

Comment:
i think we can extract this into a function because it repeats for all the components